spec/forms/state_file/az_charitable_contributions_form_spec.rb

```diff
@@ -0,0 +1,97 @@
+require "rails_helper"
+
+RSpec.describe StateFile::AzCharitableContributionsForm do
+  describe "#valid?" do
+    let(:intake) { create :state_file_az_intake }
+
+    context "with no radio selected" do
+      let(:invalid_params) do
+        {
+          charitable_contributions: "unfilled",
+        }
+      end
+
+      it "returns false" do
+        form = described_class.new(intake, invalid_params)
+        expect(form).not_to be_valid
```

Comment:
Non-blocking: when testing validation, I like to add an expectation that an error was added to the specific field I'm trying to test. For example:

```suggestion
      it "returns false" do
        form = described_class.new(intake, invalid_params)
        expect(form).not_to be_valid
        expect(form.errors).to include(:charitable_contributions)
```

This way, if the form is invalid due to a different field, or the error hasn't been properly added to the form's list of errors, then it won't pass.